pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,8 +54,13 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
 	APIVersion string `json:"api-version,omitempty"`
+	// PreferredVersions takes a string of multiple group/version pairs (separated by comma), and informs version negotiation
+	PreferredVersions string `json:"preferred-versions,omitempty"`
```

Comment:
Can we call this `PreferredGroupVersions` to avoid ambiguity and make the struct field `[]string`?  That makes the code more obvious.  I don't object to the shorter flag name, but there could be usability benefits to setting it to be a `[]string` flag as well.